### PR TITLE
Update `kedro new` prompts

### PR DESCRIPTION
**Pull request by AhdraMeraliQB**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

Fixes part of #3054:
- Switch order of prompts - 1. name 2. addons 3. example
- Adjust copy at the selection step will be updated to mention the benefits of using "software engineering best practice"

[Link to new copy](https://github.com/kedro-org/kedro/issues/3054#issuecomment-1775659975)

I believe this PR also addresses #3341 

Question: The updated copy has no mention of any message for the case where an example pipeline isn't selected. Should we add one? Also should we be leveraging colour anywhere?

## Development notes
name and add-ons take default values, example pipeline is selected (yes)
Previous flow:
image

Updated flow (as of [e7da825](https://github.com/kedro-org/kedro/pull/3355/commits/e7da82553d059000950fe51abd0b6e6dd7338384)):
image

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: kedro/templates/project/prompts.yml [hunk 1]
```diff
@@ -1,40 +1,43 @@
...
 
     Add-Ons
-    1) Linting : Provides a basic linting set up with Black and ruff
-    2) Testing : Provides basic testing set up with pytest
-    3) Custom Logging : Provides more logging options
-    4) Documentation: Provides basic documentations setup with Sphinx
-    5) Data Structure: Provides a directory structure for storing data
-    6) Pyspark: Provides set up configuration for working with PySpark
-    7) Kedro Viz: Provides Kedro's native visualisation tool
+    1) Lint: Provides a basic linting set up with Black, Ruff
```
Comment (stichbury) [inline kedro/templates/project/prompts.yml:20]:
```suggestion
    1) Lint: Provides a basic linting set up with Black, or Ruff
```

Comment (stichbury) [reply kedro/templates/project/prompts.yml:20]: "Black or Ruff" or "Black and Ruff"?

File: kedro/templates/project/prompts.yml [hunk 2]
```diff
@@ -1,40 +1,43 @@
...
+    2) Test: Provides basic testing set up with pytest
```
Comment (stichbury) [inline kedro/templates/project/prompts.yml:21]:
```suggestion
    2) Test: Basic testing set up with pytest
```

Comment (stichbury) [reply kedro/templates/project/prompts.yml:21]: Please can you remove the "Provides" in each case...? On second thoughts, maybe I'll make a commit on your branch?

Comment (AhdraMeraliQB) [reply kedro/templates/project/prompts.yml]: No worries, it's been dropped in [c5c828e](https://github.com/kedro-org/kedro/pull/3355/commits/c5c828e52daa5cb82d38510cfc44875fd62da06d)

**Comment by stichbury** [comment]
I've added some changes -- please take a look and make any further updates on top of the commit. I am sorry I messed up the DCO signoff with the change (and as I reverted it) so it's failing the DCO check at present. We can fix later once we're happy with the text.

File: kedro/framework/cli/starters.py [hunk 3]
```diff
@@ -830,34 +830,28 @@ def _create_project(template_path: str, cookiecutter_args: dict[str, Any]):
...
-    python_package = extra_context.get(
-        "python_package", project_name.lower().replace(" ", "_").replace("-", "_")
-    )
     add_ons = extra_context.get("add_ons")
+    example_pipeline = extra_context.get("example_pipeline")
+
+    click.secho(
+        "\nCongratulations!"
+        f"\nYour project '{project_name}' has been created in the directory \n{result_path}"
+    )
```
[the remark this answers is not in the document]
Comment (AhdraMeraliQB) [reply kedro/framework/cli/starters.py:839]: Any suggestions for colour? CC: @amandakys

**Comment by AhdraMeraliQB** [comment]
Closed in favour of #3358

**Pull request closed without merging**